step-templates/tomcat-undeploy-application.json

```diff
@@ -0,0 +1,73 @@
+    "Packages": [],
+    "Properties": {
+      "Octopus.Action.Script.ScriptSource": "Inline",
+      "Octopus.Action.Script.Syntax": "Bash",
+      "Octopus.Action.Script.ScriptBody": "# Get variables\ntomcatManagerUrl=$(get_octopusvariable \"UndeployManagerUrl\")\ntomcatManagementUser=$(get_octopusvariable \"UndeployManagementUser\")\ntomcatManagementPassword=$(get_octopusvariable \"UndeployManagementPassword\")\ncontextPath=$(get_octopusvariable \"UndeployContextPath\")\ndeploymentVersion=$(get_octopusvariable \"UndeployDeploymentVersion\")\napplicationFound=false\ndisplayMessage=\"$contextPath\"\n\n# Get list of applications\necho \"Checking Tomcat for $contextPath ...\"\nlistUrl=\"$tomcatManagerUrl/text/list\"\nresults=$(curl $listUrl --user \"$tomcatManagementUser\":\"$tomcatManagementPassword\" 2>&1)\n\n# Break results into an array\nIFS=$'\\n' resultArray=($results)\n\n# Loop through results\nfor i in \"${resultArray[@]}\"\ndo\n\t# Check for context path\n    if [[ \"$i\" == *\"$contextPath\"* ]]\n    then\n    \t# Check to see if there was a version specified\n        if [[ \"$deploymentVersion\" != \"\" ]]\n        then\n        \tdisplayMessage=\"$displayMessage $deploymentVersion\"\n        \t# Check for version\n            if [[ \"$i\" == *\"$deploymentVersion\"* ]]\n            then\n            \techo \"Found $contextPath with version $deploymentVersion ...\"\n                applicationFound=true\n                break\n            fi\n        else\n        \tif [[ \"$i\" != *\"##\"* ]]\n            then\n            \techo \"Found $contextPath ...\"\n            \tapplicationFound=true\n            \tbreak\n            fi\n        fi\n    fi\ndone\n\nif [[ \"$applicationFound\" == true ]]\nthen\n\t# Create URL\n\tundeployUrl=\"$tomcatManagerUrl/text/undeploy?path=/$contextPath\"\n    \n\t# Check to see if a version was specified\n\tif [[ \"$deploymentVersion\" != \"\" ]]\n\tthen\n\t\tundeployUrl=\"$undeployUrl&version=$deploymentVersion\"\n\tfi\n\n\t# Let user know what's going on\n\techo \"Removing $displayMessage ...\"\n\n\t# Call the undeploy for Tomcat\n\tcurl \"$undeployUrl\" --user \"$tomcatManagementUser\":\"$tomcatManagementPassword\" 2>&1\nelse\n\techo \"Unable to find $displayMessage ...\"\nfi"
+    },
+    "Parameters": [
+      {
+        "Id": "a3a0b5b5-067f-40fe-b93d-833bbe901937",
+        "Name": "UndeployManagerUrl",
+        "Label": "Tomcat Manager URL",
+        "HelpText": "This is the URL of the Tomcat Manager instance that the package will be uploaded to. This URL is relative to the host that is executing the step. So if the Tentacle is on the same host as Tomcat, then this value may be something like http://localhost:8080/manager.",
+        "DefaultValue": "http://localhost:8080/manager",
+        "DisplaySettings": {
+          "Octopus.ControlType": "SingleLineText"
+        }
+      },
+      {
+        "Id": "bb902de7-ad45-4f76-8111-0d080d491c77",
+        "Name": "UndeployManagementUser",
```

Comment:
I think it would be helpful to namespace these parameters. Maybe some like _Tomcat.Undeploy.ManagerUser_.